### PR TITLE
Can't delete round members

### PR DESCRIPTION
**Pull request by Powersource**
for https://github.com/cobudget/cobudget/issues/586

* [x] don't show isRemoved members in the "to approve" list
* [x] only allow removing people if they have 0 balance
  * [x] api
  * [x] ui
* [x] go through the different member-adding paths and make sure they work for a user who used to be a member
* [x] what about members with "invitation pending"

### CODE REVIEW

File: ui/components/Header/index.tsx [hunk 1]
```diff
@@ -160,36 +163,40 @@ const Header = ({ collection, currentUser, currentOrg, openModal, router }) => {
...
-                              toast.error(error.message);
-                            } else {
-                              toast.success(
-                                collection.registrationPolicy ===
-                                  "REQUEST_TO_JOIN"
-                                  ? "Request sent!"
-                                  : "You joined this collection!"
-                              );
+                {
+                  // you can ask to be a member if you've either never been a member, or been a member but been removed
```
Comment (Powersource) [inline ui/components/Header/index.tsx:167]: doublecheck the logic here before merging

Comment (Powersource) [reply ui/components/Header/index.tsx:168]: guarded a bit more now

**Comment by Powersource** [comment]
@gustavlrsn I've tried to be careful with the logic and have tested all the user flows I could think of but if you could ctrl-f through the diff for `isApproved` `hasJoined` and `isRemoved` and see that the logic makes sense then that'd be nice. It's also possible I've forgotten some case where we use these variables and we'd need to update the logic, but since I've sort of "hid" isRemoved behind isApproved (see comment in the schema) I think we should be fine.

**Comment by Powersource** [comment]
also this pr inspired https://github.com/cobudget/cobudget/issues/627

**Comment by gustavlrsn** [comment]
LGTM! This seems like the right approach 👍

**Comment by gustavlrsn** [comment]
Some merge conflicts

**Pull request merged**